### ISSUE TITLE
Failing to compile in Xcode 9

### ISSUE THREAD

**Issue by jhristov** [aleph7/Surge#78]
Hello, I've just upgraded to Xcode 9 where Upsurge fails to install - either through Carthage or cocoapods giving these compilation errors:

Upsurge/Source/LinearOperators.swift:58:15: error: generic parameter 'ML' is not used in function signature
public func +<ML: LinearType, MR: LinearType>(lhs: Double, rhs: MR) -> ValueArray<Double> where ML.Element == Double, MR.Element == Double {
              ^
Upsurge/Source/LinearOperators.swift:227:15: error: generic parameter 'ML' is not used in function signature
public func +<ML: LinearType, MR: LinearType>(lhs: Float, rhs: MR) -> ValueArray<Float> where ML.Element == Float, MR.Element == Float {
              ^
Upsurge/Source/Span.swift:22:15: error: 'Sequence' requires the types 'Span.Element' (aka 'CountableClosedRange<Int>') and '[Int]' be equivalent
public struct Span: ExpressibleByArrayLiteral, Sequence {
              ^

Upsurge/Source/Span.swift:22:15: error: 'Sequence' requires the types 'Span.Element' (aka 'CountableClosedRange<Int>') and '[Int]' be equivalent

**Comment by alejandro-isaza** [comment]
Added an xcode9 branch, give that a try.

**Comment by alldritt** [comment]
Hi,

I'm seeing these same errors as well, and the xcode9 branch does not appear to exist.

**Comment by ghost** [comment]
Try `master` or version `0.9.0`

**Comment by alldritt** [comment]
Thank you for the suggestion.  I tried both master and '0.9.0' and I continue to get a compilation error with Xcode 9.0 (9A235) on Sierra:

~~~
/Users/mall/Desktop/UpsurgeXCode9Test/Pods/Upsurge/Source/2DTensorSlice.swift:62:42: Contextual closure type '(_, _) -> _' expects 2 arguments, but 1 was used in closure body
~~~

This is better than before where there were 3 errors.

Attached is a demo project I created to test this. 

[UpsurgeXCode9Test.zip](https://github.com/aleph7/Upsurge/files/1325422/UpsurgeXCode9Test.zip)

**Comment by alejandro-isaza** [comment]
Try the `Swift4.0` branch I'll merge this in a bit.

**Comment by alldritt** [comment]
Success - thank you.

**Comment by alejandro-isaza** [comment]
Merged to master and tagged 0.10.0
